src/Plugin/views/filter/NodeHasMediaUse.php

```diff
@@ -88,7 +88,7 @@ public function query() {
     $sub_query->join('media__field_media_of', 'of', 'm.mid = of.entity_id');
     $sub_query->fields('of', ['field_media_of_target_id'])
       ->condition('use.field_media_use_target_id', $term->id());
-    $this->query->addWhere(0, 'nid', $sub_query, $condition);
+    $this->query->addWhere(0, 'node_field_data.nid', $sub_query, $condition);
```

Comment:
Not sure this should be hard-coded, as the table could be joined against multiple times? I'm thinking that the filter class could provide the appropriate alias, in some way?

Reply:
Quick bit of poking, something like:

```suggestion
    /** @var \Drupal\views\Plugin\views\query\Sql $query */
    $query = $this->query;
    $alias = $query->ensureTable('node_field_data', $this->relationship);
    $query->addWhere(0, "{$alias}.nid", $sub_query, $condition);
```

Reply:
Bit of other paranoia with the aliases here, but yeah, could pull in https://github.com/Islandora/islandora/compare/2.x...adam-vessey:islandora:fix/relationships ?

Reply:
Might've gotten a little distracted, and thrown together something of a dependency-injection pass on the class as well: https://github.com/Islandora/islandora/compare/2.x...adam-vessey:islandora:fix/dependency-injection

Reply:
oh dear. I screwed up and got a bunch of old commits. I'll try to rebase.